lms/services/async_oauth_http.py

```diff
@@ -0,0 +1,60 @@
+import asyncio
+from typing import List
+
+import aiohttp
+
+from lms.services import ExternalRequestError
+
+
+class AsyncOAuthHTTPService:
+    def __init__(self, oauth2_token_service):
+        self._oauth2_token_service = oauth2_token_service
+
+    def get(
+        self, urls, concurrency=10, headers=None, timeout=10
+    ) -> List[aiohttp.ClientResponse]:
+        """
+        Sends async requests to all urls.
+
+        :raise OAuth2TokenError: if we don't have an access token for the user
+        :raise ExternalRequestError: if something goes wrong with the HTTP
+            request
+        """
+
+        headers = headers or {}
+
+        access_token = self._oauth2_token_service.get().access_token
+        headers["Authorization"] = f"Bearer {access_token}"
+
+        async def _async_request(semaphore, aio_session, url, headers=None):
```

Comment:
I kept this as private an even inside the main `get` method to hide all the `async` bits away. They probably could live as private top level methods

Reply:
They look like they could even be private helper functions outside the class, they don't use `self`. Or private `@staticmethod`'s at the bottom of the class